Code improvements and fixes

ArtyomArles opened the pull request.

ArtyomArles committed:
Code improvements and fixes

anastasia-mironova commented on src/App.js:
```diff
@@ -1,53 +1,72 @@
-				}
-			}}>Меню</Button>
+  useEffect(() => {
+    const selectBlock = document.getElementById('selectBlock')
```
если используешь реакт, то плохая практика манипулировать дом деревом напрямую и задавать ему стили динамически
для таких случаев лучше использовать css классы у тегов и библиотеку classnames

anastasia-mironova commented on src/components/DropDownList.js:
```diff
@@ -1,26 +1,33 @@
+        setSelectValue('')
+        onChange()
+      }}
+    >Убрать фильтр</Button>
```
нужно перенести текст и тег на новые строки

anastasia-mironova commented on src/components/JustTable.js:
```diff
@@ -0,0 +1,16 @@
+
+export default function JustTable({title, columns, dataSource}) {
+
+  return (<div className='main'>
```
перенести на новую строку

anastasia-mironova commented on src/components/TableComponent.js:
```diff
@@ -1,243 +1,318 @@
+    title: 'Рег.номер',
+    dataIndex: 'regNumber',
+    key: 'regNumber',
+    render: (text) => <a href="https://ya.ru/">{text}</a>,
```
зачем ссылка на яндекс?)

anastasia-mironova commented on src/components/Profile.js:
```diff
@@ -1,18 +1,18 @@
+import React from 'react'
+import {CgProfile} from 'react-icons/cg'
 
 class Profile extends React.Component {
```
тут не поправил на функциональный компонент

anastasia-mironova commented on src/components/DropDownList.js:
```diff
@@ -1,26 +1,33 @@
 
-export default DropDownList
\ No newline at end of file
+  return (<div className="dropDownList">
```
здесь лучше  div переносить на новую строку

anastasia-mironova commented on src/components/DropDownList.js:
```diff
@@ -1,26 +1,33 @@
+      virtual={false}
+      value={selectValue}
+      options={options}
+      onPopupScroll={() => { }}
```
лишний обработчик?

anastasia-mironova commented on src/components/SelectElement.js:
```diff
@@ -1,24 +1,22 @@
 
-export default SelectElement
\ No newline at end of file
+  return (<div className="selectElement"
```
перенеси div на новую строку

anastasia-mironova commented on src/components/SelectElement.js:
```diff
@@ -1,24 +1,22 @@
+      const el1 = document.getElementById('1')
+      el1.classList.remove('currentTable')
+      const el2 = document.getElementById('2')
+      el2.classList.remove('currentTable')
```
лучше здесь тоже уйти от манипулирования дом

ArtyomArles committed:
Fix comments

popovalxndr commented on src/components/TableComponent.js:
Нужно пересмотреть подход к формированию таблицы - не завязываться на конкретные id справочников, а обеспечить более гибкий вывод информации, который позволит в случае изменения json с данными справочника выводить данные без доработок данного компонента.

popovalxndr reviewed (commented):
Прошу выполнить следующие доработки:
1) Попробовать сделать минимизацию меню вбок влево
2) Реализовать сохранение поисковых параметров, чтобы фильтры оставались заполненными после перезагрузки страницы (варианты решения - query-string, redux-persist, local storage)
3) Фильтры по ЦОДу и Статусу попробовать сделать множественными
4) Для признака репликации в фильтре, а также в реестре выводить Да/Нет вместо true/false
5) Сделать кнопку очистки всех фильтров сразу
6) Для дропдаунов фильтров попробовать реализовать очистку библиотечным способом (крестик в самом дропдауне, а не отдельной кнопкой)
7) Добавить к поиску дебаунсинг - замечание актуально
8) В справочники добавить строку поиска для поиска по коду или наименованию
9) При открытии http://localhost:3000/ исправить варнинг в консоли: [Warning] No routes matched location "/"
10) Доправить варнинг по key в консоли браузера
11) Реализовать открытие формы потребности по клику на рег. номер в таблице - сейчас открывается Яндекс=)
12) Для справочников также реализовать формы
13) Поэкспериментировать с redux - например, загружать в него данные справочников и брать их из redux для отображения

ArtyomArles committed:
fixing bugs, creating a pop-up modal and connecting local storage

ArtyomArles committed:
creating a separate Filters component, restoring debouncing and minor code improvements

ArtyomArles committed:
Fixing bugs, adding keys to data, renaming a component from TableComponent to PreordersTable

anastasia-mironova commented on src/App.js:
```diff
@@ -1,53 +1,39 @@
-}
-export default App
\ No newline at end of file
+  return (<div className='main'>
```
Перенеси тег на новую строку

anastasia-mironova commented on src/components/ListMenu.js:
```diff
@@ -0,0 +1,72 @@
+  const menuKeyPath = useSelector(state => state.memory.menuKeyPath)
+  const dispatch = useDispatch()
+
+  return (<div className='menu'>
```
Перенеси тег на новую строку

anastasia-mironova commented on src/components/PreordersTable.js:
```diff
@@ -0,0 +1,243 @@
+    </div>
+    <Routes>
+      <Route path={selectedPreorder.regNumber} element={
+        <Modal
```
если у компонента больше одного пропа, то каждый переносится на новую строку

anastasia-mironova commented on src/components/Profile.js:
```diff
@@ -1,18 +1,10 @@
-export default Profile
\ No newline at end of file
+export default function Profile({name}) {
+  return (<div className="profile">
```
Перенеси тег на новую строку

anastasia-mironova commented on src/store/editFilter.js:
```diff
@@ -0,0 +1,78 @@
+      state.filter.statuses = action.payload
+    },
+    clearFilter(state) {
+      state.filter = {
```
вынеси filter из initial state в отдельный объект и используй его при инициализации и очистке

anastasia-mironova commented on src/store/memory.js:
```diff
@@ -0,0 +1,39 @@
+    },
+    editFiltersCollapsed(state) {
+      if (state.filtersCollapsed === undefined) {
+        state.filtersCollapsed = {display: 'flex'}
```
плохой подход с добавлением инлайн стилей в стор 
в стор лучше добавить флаг и в зависимости от его значений добавить css в компоненте

anastasia-mironova replied:
и вообще, тут правильнее будет скорее вот так:
const initialState = {filtersCollapsed: false}
state.filtersCollapsed = !state.filtersCollapsed 

{filtersCollapsed && 
       <Filter />}

ArtyomArles committed:
pre-final version

popovalxndr reviewed (commented):
Остались актуальными следующие замечания:
- Добавить к поиску дебаунсинг;
- При открытии http://localhost:3000/ исправить варнинг в консоли: [Warning] No routes matched location "/";
- Для справочников также реализовать формы;
- Нужно пересмотреть подход к формированию таблицы - не завязываться на конкретные id справочников, а обеспечить более гибкий вывод информации, который позволит в случае изменения json с данными справочника выводить данные без доработок данного компонента.

ArtyomArles committed:
Bug fixes other than debounce

Commit:
Добавил дебаунсинг и использовал в Потребностях

popovalxndr reviewed (commented):
Предлагаю еще одну доработку: если находишься на форме потребности или справочника и перезагружаешь страницу, то открывать эту же форму, а не реестр.

popovalxndr commented on src/components/PreordersTable.js:
```diff
@@ -0,0 +1,230 @@
+      key: 'preorderTypeId',
+      width: 150,
+      render: (text) => {
+        if (text === 1) {
```
Вот про код такого вида здесь и ниже имелся в виду комментарий:
"Нужно пересмотреть подход к формированию таблицы - не завязываться на конкретные id справочников, а обеспечить более гибкий вывод информации, который позволит в случае изменения json с данными справочника выводить данные без доработок данного компонента."

ArtyomArles committed:
Добавление объектов в локальное хранилище

Добавил дополнительные объекты в локальное хранилище для реализации функции сохранения открытия модального окна при обновлении страницы

ArtyomArles committed:
Merge branch 'develop' of https://github.com/ArtyomArles/practice into develop

ArtyomArles committed:
Изменение кода формирования таблицы
